docs/src/main/asciidoc/_developer-guide/JPA_Reference_Guide.adoc

```diff
@@ -731,10 +668,89 @@ import org.hibernate.SessionFactory;
 }
 ----
 
+[[hibernate-transformer]]
+== Hibernate ORM 5.1 native API bytecode transformer
+The *Hibernate ORM 5.1 native API bytecode transformer*, can rewrite application bytecode,
+to handle APIs that have changed between Hibernate ORM 5.1.x + 5.3.x.  For applications that use the Hibernate ORM 5.1
+native APIs, you may either rewrite them for Hibernate ORM 5.3 or use the transformer, to rewrite your application byte-code
+during application deployment (only the in-memory copy of your application is rewritten).  If your application is only using the
+JPA specification APIs, you do not need to transform your application.
+
+For details on the Hibernate 5.3 native API changes, review the https://github.com/hibernate/hibernate-orm/blob/5.2/migration-guide.adoc[ORM 5.2]
++ https://github.com/hibernate/hibernate-orm/blob/5.3/migration-guide.adoc[ORM 5.3] migration guides.
+
+The Hibernate ORM 5.1 transformer is already deprecated, meaning it could be removed in the next release.
+For this reason, we made it easy to enable the transformer via either a system property or jboss-deployment-structure change.
+
+[source,xml]
+----
+<jboss-deployment-structure>
+    <deployment>
+        <transformers>
+            <transformer class="org.jboss.as.hibernate.Hibernate51CompatibilityTransformer"/>
+        </transformers>
+        <dependencies>
+            <module name="org.hibernate" export="true" />
+        </dependencies>
+    </deployment>
+    <sub-deployment name="main.war">
+        <transformers>
+            <transformer class="org.jboss.as.hibernate.Hibernate51CompatibilityTransformer"/>
+        </transformers>
+    </sub-deployment>
+</jboss-deployment-structure>
+----
+
+
+If enabled, the following transformations will be made to application classes compiled with Hibernate 5.1: ::
+*  In user type classes, change methods with parameter type org.hibernate.engine.spi.SessionImplementor to instead use type org.hibernate.engine.spi.SharedSessionContractImplementor, this is done for implementations of:
+**  org.hibernate.usertype.UserType.
+**  org.hibernate.usertype.CompositeUserType.
+**  org.hibernate.usertype.UserCollectionType.
+**  org.hibernate.usertype.UserVersionType.
+**  org.hibernate.type.Type.
+**  org.hibernate.type.SingleColumnType.
+**  org.hibernate.type.AbstractStandardBasicType.
+**  org.hibernate.type.ProcedureParameterExtractionAware.
+**  org.hibernate.type.ProcedureParameterNamedBinder.
+**  org.hibernate.type.VersionType.
+*  In user type classes, when calling hibernate.orm methods, cast SessionImplementor to SharedSessionContractImplementor.
```

Comment:
s/hibernate.orm/Hibernate ORM/? Or s/hibernate.orm/org.hibernate/?

hibernate.orm doesn't really have a meaning IMHO.

Reply:
thanks, definitely was a typo!  Fixing...

Reply:
Changing to:  In user type classes, when calling org.hibernate.* class methods, cast SessionImplementor to SharedSessionContractImplementor.